test/unit/queue/src/QueueTest.cpp

```diff
@@ -171,3 +172,34 @@ TEMPLATE_LIST_TEST_CASE("queueShouldNotExecuteTasksInParallel", "[queue]", TestQ
     firstTaskFinishedFuture.get();
     secondTaskFinishedFuture.get();
 }
+
+//! This task launches a long task in a non-blocking queue and destroys the
+//! queue before the task is finished. The run time of the task is a bit longer
+//! than all other tests here, so the task would run past program termination if
+//! not synchronized somewhere. This test has no assertion, because the tested
+//! behviour is outside of any function scope.
+//!
+//! A thread running past program termination invoked undefined behavior, which
+//! occasionally leads to crashes after termination. I.e. this test may crash
+//! after catch2 reported success.
```

Comment:
Would a catch2 check at the end of the long task be useful in marking the "undefined behaviour" as an explicit failure ?

Reply:
No, because that check would just never happen in the failure scenario: The thread runs and either causes some trouble, because it touches memory or calls a function related to something already cleaned-up after `main()` or has no effect and at some point just gets killed by the OS when parent process terminates. So, the thread would not get to that check. If we were to wait for the check, then we eliminate the failure scenario, by effectively joining that thread.

It would be better if we could check for some defined behavior instead of checking for nothing undefined happening. For example, if in alpaka all queues devices were required to synchronize with and destroy their attached queues in their destructor, we could check that. But this is not a thing that alpaka devices do.